server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -71,25 +68,37 @@
         public static final BytesRef FALSE = new BytesRef("F");
     }
 
-    public static class Builder extends FieldMapper.Builder<Builder> {
+    private static BooleanFieldMapper toType(FieldMapper in) {
+        return (BooleanFieldMapper) in;
+    }
+
+    public static class Builder extends ParametrizedFieldMapper.Builder {
 
-        private Boolean nullValue;
+        private final Parameter<Float> boost = Parameter.floatParam("boost", true, m -> m.fieldType().boost(), 1.0f);
+        private final Parameter<Boolean> docValues
+            = Parameter.boolParam("doc_values", false, m -> m.fieldType().hasDocValues(),  true);
+        private final Parameter<Boolean> indexed
+            = Parameter.boolParam("index", false, m -> m.fieldType().isSearchable(), true);
+        private final Parameter<Boolean> nullValue
+            = new Parameter<>("null_value", false, null, (n, o) -> XContentMapValues.nodeBooleanValue(o), m -> toType(m).nullValue);
+        private final Parameter<Boolean> stored = Parameter.boolParam("store", false, m -> toType(m).stored, false);
+        private final Parameter<Map<String, String>> meta
+            = new Parameter<>("meta", true, Collections.emptyMap(), TypeParsers::parseMeta, m -> m.fieldType().meta());
```

Comment:
I get confused here  that stored seems to be treated differently from e.g. doc_values. Why is that?

Reply:
Convenience, mainly - `doc_values` and `index` can be derived from the field type values, but `stored` can't be.  I can change it so that `doc_values` and `index` are fields directly on the mapper as well if you think that's clearer?  Certainly for things like searchable docvalues fields then we'll need to distinguish between whether or not something is indexed and whether it's searchable.

Reply:
I guess I am trying to see what the difference is between stored and indexed/doc_values. I would expect them to be treated in the same way/same place. Why can't stored be derived from the field type values?

Reply:
As it stands, `MappedFieldType` doesn't have anything that tells you if the field is stored or not.

Reply:
You made a `boolean hasDocValues` on `BinaryFieldMapper` for this. Is it worth making one here too?

Reply:
I'll add `indexed` and `hasDocValues` directly to the mappers as it sounds as though that will be less confusing.